### PR TITLE
speedb: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **speedb/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/speedb/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        tc.variables["WITH_GFLAGS"] = self.options.with_core_tools
+        tc.variables["WITH_TOOLS"] = False
+        tc.variables["ROCKSDB_BUILD_SHARED"] = self.options.shared
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # to avoid wrong CPU detection, modify not to use "-march=native" in apple-clang
```
Comment (uilianries) [inline recipes/speedb/all/conanfile.py:114]: @toge Is it to be able to compiler in Mac M1? If yes, could you please report the upstream about the case?

File: recipes/speedb/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["speedb"]
+        self.cpp_info.set_property("pkg_config_name", "speedb")
```
Comment (uilianries) [inline recipes/speedb/all/conanfile.py:138]:
```suggestion
        self.cpp_info.set_property("pkg_config_name", "speedb")
        self.cpp_info.set_property("cmake_file_name", "Speedb")
        self.cpp_info.set_property("cmake_target_name", "Speedb::speedb")
```

https://github.com/speedb-io/speedb/blob/main/CMakeLists.txt#L1287
https://github.com/speedb-io/speedb/blob/main/CMakeLists.txt#L1337

**Review by uilianries — COMMENTED** [review]
I did not understand very this project, is it a wrapper to Rocksdb or is it vendorized in the project? I just build Rocksdb and the generated headers files are the same! So or this recipe should use `provide = "rocksdb"` or check is using rocksdb as vendor and we could (or not) separate it.

**Comment by toge** [comment]
@uilianries 
Since speedb has become a library featuring 100% compatibility with rocksdb, it seems that the header file names are unified. Although it features compatibility, it does not seem to be a wrapper for rocksdb.

File: recipes/speedb/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        tc.variables["WITH_SNAPPY"] = self.options.with_snappy
+        tc.variables["WITH_LZ4"] = self.options.with_lz4
+        tc.variables["WITH_ZLIB"] = self.options.with_zlib
+        tc.variables["WITH_ZSTD"] = self.options.with_zstd
+        tc.variables["WITH_TESTS"] = False
+        tc.variables["WITH_BENCHMARK_TOOLS"] = False
+        tc.variables["WITH_CORE_TOOLS"] = self.options.with_core_tools
+        tc.variables["WITH_GFLAGS"] = self.options.with_core_tools
+        tc.variables["WITH_TOOLS"] = False
+        tc.variables["ROCKSDB_BUILD_SHARED"] = self.options.shared
```
Comment (valgur) [inline recipes/speedb/all/conanfile.py:107]:
```suggestion
        tc.variables["ROCKSDB_BUILD_SHARED"] = self.options.shared
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```

File: recipes/speedb/all/patches/2.8.0-0001-use-cci.patch
Comment (valgur) [inline recipes/speedb/all/patches/2.8.0-0001-use-cci.patch:1]: Can be avoided by using `CMakeDeps.set_property()` on `cmake_file_name` and `cmake_target_name` properties.

**Review by valgur — CHANGES_REQUESTED** [review]
CMP0077 is required, otherwise LGTM.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**